postgres/datadog_checks/postgres/statements.py

```diff
@@ -0,0 +1,181 @@
+            logger.warning('Statement-level metrics are unavailable: %s', e)
+            return []
+
+    def _get_pg_stat_statements_columns(self, db):
+        """
+        Load the list of the columns available under the `pg_stat_statements` table. This must be queried because
+        version is not a reliable way to determine the available columns on `pg_stat_statements`. The database can
+        be upgraded without upgrading extensions, even when the extension is included by default.
+        """
+        query = """
+            SELECT column_name
+            FROM information_schema.columns
+            WHERE table_schema = 'public'
+            AND table_name = 'pg_stat_statements';
+            """
+        columns = self._execute_query(db.cursor(), query)
+        return [column[0] for column in columns]
+
+    def collect_per_statement_metrics(self, instance, db, instance_tags):
+        try:
+            self.__collect_per_statement_metrics(instance, db, instance_tags)
+        except Exception:
+            db.rollback()
+            logger.exception('Unable to collect statement metrics due to an error')
+
+    def __collect_per_statement_metrics(self, instance, db, instance_tags):
+        available_columns = self._get_pg_stat_statements_columns(db)
+        missing_columns = PG_STAT_STATEMENTS_REQUIRED_COLUMNS - set(available_columns)
+        if len(missing_columns) > 0:
+            logger.warning(
+                'Unable to collect statement metrics because required fields are unavailable: %s',
+                ', '.join(list(missing_columns)),
+            )
+            return
+
+        desired_columns = (
+            list(PG_STAT_STATEMENTS_METRIC_COLUMNS.keys())
+            + list(PG_STAT_STATEMENTS_OPTIONAL_COLUMNS)
+            + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        query_columns = list(
+            set(desired_columns) & set(available_columns) + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        rows = self._execute_query(
+            db.cursor(cursor_factory=psycopg2.extras.DictCursor),
+            STATEMENTS_QUERY.format(
+                cols=', '.join(query_columns),
+                pg_stat_statements_view=self.config.pg_stat_statements_view,
+            ),
+            params=(self.config.dbname,),
+        )
+        if not rows:
+            return
+
+        def row_keyfunc(row):
+            # old versions of pg_stat_statements don't have a query ID so fall back to the query string itself
+            queryid = row['queryid'] if 'queryid' in row else row['query']
+            return (queryid, row['datname'], row['rolname'])
+
+        rows = self._state.compute_derivative_rows(rows, PG_STAT_STATEMENTS_METRIC_COLUMNS.keys(), key=row_keyfunc)
+        metric_limits = (
+            self.config.statement_metric_limits if self.config.statement_metric_limits else DEFAULT_METRIC_LIMITS
+        )
```

Comment:
I think we should move the "which default should we use?" logic to `config.py`, right when we get the option:

```python
self.statement_metric_limits = instance.get('statement_metric_limits', DEFAULT_METRIC_LIMITS)
```

Here: https://github.com/DataDog/integrations-core/pull/7852/files#diff-6a577b5243942f4892d04e92192fa51758a4375449ee63d0286306e988bbfac2R66

And then we can simplify this to:

```suggestion
        metric_limits = self.config.statement_metric_limits
```

(Separation of concerns, etc.)

Reply:
Removed as a configurable limit